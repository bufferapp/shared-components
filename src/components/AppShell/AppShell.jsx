import React from 'react';
import PropTypes from 'prop-types';
import NavBar from '../NavBar';
// import GlobalStyles from '../GlobalStyles';

import {
  AppShellStyled,
  Wrapper,
  SidebarWrapper,
  ContentWrapper,
} from './style';

/**
 * The AppShell component is a general purpose wrapper for all of our applications. At the moment it's primarily a wrapper for the `NavBar` component. Check out the example below to see how to integrate it into your app.
 */
const AppShell = ({ activeProduct, user, helpMenuItems, sidebar, content }) => (
  <AppShellStyled>
    {/* <GlobalStyles /> */}
    <NavBar activeProduct={activeProduct} user={user} helpMenuItems={helpMenuItems} />
    <Wrapper>
      {sidebar && <SidebarWrapper>{sidebar}</SidebarWrapper>}
      <ContentWrapper>{content}</ContentWrapper>
    </Wrapper>
  </AppShellStyled>
);

AppShell.propTypes = {
  /** The currently active (highlighted) product in the `NavBar`, one of `'publish', 'reply', 'analyze'` */
  activeProduct: PropTypes.oneOf(['publish', 'reply', 'analyze']),

  /** See the prop description in the `NavBar` component. */
  user: NavBar.propTypes.user, // eslint-disable-line

  /** See the prop description in the `NavBar` component. */
  helpMenuItems: NavBar.propTypes.helpMenuItems, // eslint-disable-line

  /** (Optional) Your sidebar component. */
  sidebar: PropTypes.node,

  /** (Optional) Your content component. */
  content: PropTypes.node.isRequired,
};

AppShell.defaultProps = {
  sidebar: null,
  activeProduct: undefined,
}

export default AppShell;
