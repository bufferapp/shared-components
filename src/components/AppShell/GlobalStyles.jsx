import { createGlobalStyle } from 'styled-components';

/**
 * https://www.styled-components.com/docs/api#createglobalstyle
 */
const GlobalStyles = createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    font-family: "Roboto", sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }
`;

export default GlobalStyles;
