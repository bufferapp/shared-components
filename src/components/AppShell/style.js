import styled, { css } from 'styled-components';
import { grayLighter } from '../style/colors';

const flexColumn = css`
  display: flex;
  flex-direction: column;
`;

const flexRow = css`
  display: flex;
  flex-direction: row;
`;

export const AppShellStyled = styled.div`
  ${flexColumn}
  height: 100vh;
  overflow: auto;
  background-color: ${grayLighter};
`;

export const Wrapper = styled.div`
  ${flexRow}
  flex: 1;
  overflow: auto;
`;

export const SidebarWrapper = styled.div`
  ${flexColumn}
  width: 248px;
`;

export const ContentWrapper = styled.div`
  ${flexColumn}
  align-items: center;
  flex: 1;
  overflow: auto;
`;
