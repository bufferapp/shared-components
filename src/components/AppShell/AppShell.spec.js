/* eslint-disable react/jsx-filename-extension */
import snap from 'jest-auto-snapshots';
import 'jest-styled-components';

import AppShell from './AppShell';

snap(AppShell, './AppShell.jsx');
