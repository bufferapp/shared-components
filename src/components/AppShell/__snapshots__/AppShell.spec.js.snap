// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > AppShell Matches snapshot when array prop "helpMenuItems" has one item: "[object Object]" 1`] = `
<ForwardRef(styled.div)>
  <NavBar
    activeProduct="publish"
    helpMenuItems={
      Array [
        Object {
          "component": <NodeFixture />,
          "hasDivider": true,
          "id": "jest-auto-snapshots String Fixture",
          "onItemClick": [MockFunction],
          "title": "jest-auto-snapshots String Fixture",
        },
      ]
    }
    onLogout={[MockFunction]}
    user={
      Object {
        "avatar": "jest-auto-snapshots String Fixture",
        "email": "jest-auto-snapshots String Fixture",
        "ignoreMenuItems": Array [
          "jest-auto-snapshots String Fixture",
        ],
        "menuItems": Array [
          Object {
            "component": [MockFunction],
            "hasDivider": true,
            "id": "jest-auto-snapshots String Fixture",
            "onItemClick": [MockFunction],
            "title": "jest-auto-snapshots String Fixture",
          },
        ],
        "name": "jest-auto-snapshots String Fixture",
      }
    }
  />
  <Banner
    actionButton={
      Object {
        "action": [MockFunction],
        "label": "jest-auto-snapshots String Fixture",
      }
    }
    customHTML={
      Object {
        "__html": "jest-auto-snapshots String Fixture",
      }
    }
    onCloseBanner={null}
    text="jest-auto-snapshots String Fixture"
    themeColor="blue"
  />
  <ForwardRef(styled.div)>
    <ForwardRef(styled.div)>
      <NodeFixture />
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)>
      <NodeFixture />
    </ForwardRef(styled.div)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > AppShell Matches snapshot when array prop "helpMenuItems" is an empty array: "[]" 1`] = `
<ForwardRef(styled.div)>
  <NavBar
    activeProduct="publish"
    helpMenuItems={Array []}
    onLogout={[MockFunction]}
    user={
      Object {
        "avatar": "jest-auto-snapshots String Fixture",
        "email": "jest-auto-snapshots String Fixture",
        "ignoreMenuItems": Array [
          "jest-auto-snapshots String Fixture",
        ],
        "menuItems": Array [
          Object {
            "component": [MockFunction],
            "hasDivider": true,
            "id": "jest-auto-snapshots String Fixture",
            "onItemClick": [MockFunction],
            "title": "jest-auto-snapshots String Fixture",
          },
        ],
        "name": "jest-auto-snapshots String Fixture",
      }
    }
  />
  <Banner
    actionButton={
      Object {
        "action": [MockFunction],
        "label": "jest-auto-snapshots String Fixture",
      }
    }
    customHTML={
      Object {
        "__html": "jest-auto-snapshots String Fixture",
      }
    }
    onCloseBanner={null}
    text="jest-auto-snapshots String Fixture"
    themeColor="blue"
  />
  <ForwardRef(styled.div)>
    <ForwardRef(styled.div)>
      <NodeFixture />
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)>
      <NodeFixture />
    </ForwardRef(styled.div)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > AppShell Matches snapshot when passed all props 1`] = `
<ForwardRef(styled.div)>
  <NavBar
    activeProduct="publish"
    helpMenuItems={
      Array [
        Object {
          "component": <NodeFixture />,
          "hasDivider": true,
          "id": "jest-auto-snapshots String Fixture",
          "onItemClick": [MockFunction],
          "title": "jest-auto-snapshots String Fixture",
        },
      ]
    }
    onLogout={[MockFunction]}
    user={
      Object {
        "avatar": "jest-auto-snapshots String Fixture",
        "email": "jest-auto-snapshots String Fixture",
        "ignoreMenuItems": Array [
          "jest-auto-snapshots String Fixture",
        ],
        "menuItems": Array [
          Object {
            "component": [MockFunction],
            "hasDivider": true,
            "id": "jest-auto-snapshots String Fixture",
            "onItemClick": [MockFunction],
            "title": "jest-auto-snapshots String Fixture",
          },
        ],
        "name": "jest-auto-snapshots String Fixture",
      }
    }
  />
  <Banner
    actionButton={
      Object {
        "action": [MockFunction],
        "label": "jest-auto-snapshots String Fixture",
      }
    }
    customHTML={
      Object {
        "__html": "jest-auto-snapshots String Fixture",
      }
    }
    onCloseBanner={null}
    text="jest-auto-snapshots String Fixture"
    themeColor="blue"
  />
  <ForwardRef(styled.div)>
    <ForwardRef(styled.div)>
      <NodeFixture />
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)>
      <NodeFixture />
    </ForwardRef(styled.div)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`jest-auto-snapshots > AppShell Matches snapshot when passed only required props 1`] = `
<ForwardRef(styled.div)>
  <NavBar
    helpMenuItems={null}
    user={
      Object {
        "avatar": "jest-auto-snapshots String Fixture",
        "email": "jest-auto-snapshots String Fixture",
        "ignoreMenuItems": Array [
          "jest-auto-snapshots String Fixture",
        ],
        "menuItems": Array [
          Object {
            "component": [MockFunction],
            "hasDivider": true,
            "id": "jest-auto-snapshots String Fixture",
            "onItemClick": [MockFunction],
            "title": "jest-auto-snapshots String Fixture",
          },
        ],
        "name": "jest-auto-snapshots String Fixture",
      }
    }
  />
  <ForwardRef(styled.div)>
    <ForwardRef(styled.div)>
      <NodeFixture />
    </ForwardRef(styled.div)>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;
